src/main/scala/com/fulcrumgenomics/fasta/UpdateFastaContigNames.scala

```diff
@@ -24,70 +24,157 @@
 
 package com.fulcrumgenomics.fasta
 
+import java.io.BufferedWriter
+
 import com.fulcrumgenomics.FgBioDef.PathToSequenceDictionary
 import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
 import com.fulcrumgenomics.commons.CommonsDef._
 import com.fulcrumgenomics.commons.util.LazyLogging
 import com.fulcrumgenomics.sopt.{arg, clp}
 import com.fulcrumgenomics.util.{Io, ProgressLogger}
-import htsjdk.samtools.reference.{FastaSequenceIndex, ReferenceSequenceFileFactory}
+import htsjdk.samtools.reference.{FastaSequenceIndex, ReferenceSequence, ReferenceSequenceFile, ReferenceSequenceFileFactory}
 
 @clp(description =
   """
     |Updates the sequence names in a FASTA.
     |
     |The name of each sequence must match one of the names (including aliases) in the given sequence dictionary.  The
     |new name will be the primary (non-alias) name in the sequence dictionary.
+    |
+    |By default, the sort order of the contigs will be the same as the input FASTA.  Use the `--sort` option to sort by
+    |the input sequence dictionary.  Furthermore, the sequence dictionary may contain **more** contigs than the input
+    |FASTA, and they wont be used.  Use the `--skip-missing` option to skip contigs in the input FASTA that cannot be
+    |renamed (i.e. who are not present in the input sequence dictionary).  Finally, use the `--default-contigs` to
+    |append contigs not present in the input FASTA but present in the sequence dictionary.
   """,
   group = ClpGroups.Fasta)
 class UpdateFastaContigNames
 (@arg(flag='i', doc="Input FASTA.") val input: PathToFasta,
  @arg(flag='d', doc="The path to the sequence dictionary with contig aliases.") val dict: PathToSequenceDictionary,
  @arg(flag='o', doc="Output FASTA.") val output: PathToFasta,
  @arg(flag='l', doc="Line length or sequence lines.") val lineLength: Int = 100,
- @arg(doc="Skip missing source contigs.") val skipMissing: Boolean = false
+ @arg(doc="Skip missing source contigs.") val skipMissing: Boolean = false,
+ @arg(doc="Sort the contigs based on the input sequence dictionary") sort: Boolean = false,
```

Comment:
Possibly rename to `sortByDict`. Since `--sort` usually means "output is sorted with this flag, and not sorted without this flag", and that is not the case here.